Refactor Document.h to reduce build times

jernoble opened the pull request:
#### a4a0ae0d75768887423c88d42cb963ec2035960b
<pre>
Refactor Document.h to reduce build times
<a href="https://bugs.webkit.org/show_bug.cgi?id=250887">https://bugs.webkit.org/show_bug.cgi?id=250887</a>
rdar://104465840

Reviewed by Darin Adler.

Replace explicit header includes with forward declarations wherever possible.

For some headers,
this requires enumerations to be moved outside an owning class and into namespace scope so those
enumerations can be forward declared.

For CanvasBase.h, pull CanvasObserver.h into its own file with minimal header dependencies.

For ScriptController, eliminate the requirement for SecurityOrigin to be included by modifying
executeJavaScriptURL() to take an r-value reference to a RefPtr&lt;SecurityOrigin&gt;.

With these changes, the compile-time cost of including Document.h has been reduced by 75%.

* Source/WebCore/Modules/audiosession/DOMAudioSession.h:
* Source/WebCore/Modules/mediastream/CanvasCaptureMediaStreamTrack.h:
* Source/WebCore/Modules/screen-wake-lock/WakeLock.cpp:
* Source/WebCore/Modules/webxr/WebXRSession.cpp:
* Source/WebCore/Modules/webxr/WebXRWebGLLayer.h:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/bindings/js/ScriptController.cpp:
(WebCore::ScriptController::executeJavaScriptURL):
* Source/WebCore/bindings/js/ScriptController.h:
(WebCore::ScriptController::executeJavaScriptURL):
* Source/WebCore/dom/Document.cpp:
(WebCore::Document::setFocusedElement):
(WebCore::Document::hasListenerTypeForEventType const):
(WebCore::m_reportingScope): Deleted.
* Source/WebCore/dom/Document.h:
(WebCore::Document::setFocusedElement): Deleted.
(WebCore::Document::wasLastFocusByClick const): Deleted.
(WebCore::Document::clientOrigin const): Deleted.
* Source/WebCore/dom/DocumentInlines.h:
(WebCore::Document::clientOrigin const):
(WebCore::Document::wasLastFocusByClick const):
* Source/WebCore/dom/Element.cpp:
(WebCore::isForceEvent):
(WebCore::Element::dispatchMouseForceWillBegin):
* Source/WebCore/dom/FragmentScriptingPermission.h:
* Source/WebCore/dom/KeyboardEvent.cpp:
(WebCore::eventTypeForKeyboardEventType):
* Source/WebCore/dom/VisibilityState.h:
* Source/WebCore/html/CanvasBase.h:
(WebCore::CanvasObserver::isStyleCanvasImage const): Deleted.
* Source/WebCore/html/CanvasObserver.h: Copied from Source/WebCore/page/DisabledAdaptations.h.
(WebCore::CanvasObserver::isStyleCanvasImage const):
* Source/WebCore/page/Chrome.h:
* Source/WebCore/page/ContextMenuController.cpp:
(WebCore::ContextMenuController::showContextMenuAt):
* Source/WebCore/page/DebugPageOverlays.cpp:
(WebCore::InteractionRegionOverlay::mouseEvent):
* Source/WebCore/page/DisabledAdaptations.h:
* Source/WebCore/page/DragController.cpp:
(WebCore::createMouseEvent):
* Source/WebCore/page/EventHandler.cpp:
(WebCore::SyntheticTouchPoint::SyntheticTouchPoint):
(WebCore::SyntheticSingleTouchEvent::SyntheticSingleTouchEvent):
(WebCore::EventHandler::handleMouseForceEvent):
(WebCore::EventHandler::handlePasteGlobalSelection):
(WebCore::EventHandler::sendContextMenuEventForKey):
(WebCore::EventHandler::fakeMouseMoveEventTimerFired):
(WebCore::EventHandler::isKeyEventAllowedInFullScreen const):
(WebCore::EventHandler::internalKeyEvent):
(WebCore::EventHandler::handleDrag):
(WebCore::EventHandler::dispatchSyntheticTouchEventIfEnabled):
* Source/WebCore/page/ResourceUsageOverlay.cpp:
(WebCore::ResourceUsageOverlay::mouseEvent):
* Source/WebCore/page/ios/EventHandlerIOS.mm:
(WebCore::EventHandler::dispatchSimulatedTouchEvent):
(WebCore::EventHandler::tryToBeginDragAtPoint):
* Source/WebCore/page/mac/ImageOverlayControllerMac.mm:
(WebCore::ImageOverlayController::platformHandleMouseEvent):
* Source/WebCore/page/mac/ServicesOverlayController.mm:
(WebCore::ServicesOverlayController::mouseEvent):
* Source/WebCore/platform/KeyboardScrollingAnimator.cpp:
(WebCore::keyboardScrollingKeyForKeyboardEvent):
* Source/WebCore/platform/PlatformEvent.h:
(WebCore::PlatformEvent::type const):
(WebCore::PlatformEvent::PlatformEvent):
(): Deleted.
* Source/WebCore/platform/PlatformKeyboardEvent.h:
(WebCore::PlatformKeyboardEvent::PlatformKeyboardEvent):
* Source/WebCore/platform/PlatformMouseEvent.h:
(WebCore::PlatformMouseEvent::PlatformMouseEvent):
* Source/WebCore/platform/PlatformWheelEvent.cpp:
(WebCore::PlatformWheelEvent::createFromGesture):
* Source/WebCore/platform/PlatformWheelEvent.h:
(WebCore::PlatformWheelEvent::PlatformWheelEvent):
* Source/WebCore/platform/ios/KeyEventIOS.mm:
(WebCore::PlatformKeyboardEvent::disambiguateKeyDownEvent):
* Source/WebCore/platform/ios/PlatformEventFactoryIOS.mm:
(WebCore::mouseEventType):
(WebCore::PlatformWheelEventBuilder::PlatformWheelEventBuilder):
(WebCore::PlatformKeyboardEventBuilder::PlatformKeyboardEventBuilder):
(WebCore::touchEventType):
(WebCore::touchPhaseFromPlatformEventType):
* Source/WebCore/platform/ios/ScrollAnimatorIOS.mm:
(WebCore::ScrollAnimatorIOS::handleTouchEvent):
* Source/WebCore/platform/libwebrtc/LibWebRTCVPXVideoDecoder.cpp:
* Source/WebCore/platform/mac/KeyEventMac.mm:
(WebCore::PlatformKeyboardEvent::disambiguateKeyDownEvent):
* Source/WebCore/platform/mac/PlatformEventFactoryMac.mm:
(WebCore::mouseEventTypeForEvent):
(WebCore::PlatformMouseEventBuilder::PlatformMouseEventBuilder):
(WebCore::PlatformWheelEventBuilder::PlatformWheelEventBuilder):
(WebCore::PlatformKeyboardEventBuilder::PlatformKeyboardEventBuilder):
* Source/WebCore/rendering/style/StyleCanvasImage.h:
* Source/WebCore/rendering/style/StyleCrossfadeImage.h:
* Source/WebCore/xml/DOMParser.cpp:
* Source/WebKit/Shared/WebEventConversion.cpp:
(WebKit::platform):
(WebKit::WebKit2PlatformMouseEvent::WebKit2PlatformMouseEvent):
(WebKit::WebKit2PlatformWheelEvent::WebKit2PlatformWheelEvent):
(WebKit::WebKit2PlatformKeyboardEvent::WebKit2PlatformKeyboardEvent):
(WebKit::WebKit2PlatformTouchEvent::WebKit2PlatformTouchEvent):
(WebKit::WebKit2PlatformGestureEvent::WebKit2PlatformGestureEvent):
* Source/WebKit/UIProcess/RemoteLayerTree/mac/ScrollerPairMac.mm:
(WebKit::ScrollerPairMac::handleMouseEvent):
* Source/WebKit/WebProcess/Network/WebSocketChannel.cpp:
* Source/WebKit/WebProcess/WebPage/ios/WebPageIOS.mm:
(WebKit::dispatchSyntheticMouseMove):
(WebKit::WebPage::generateSyntheticEditingCommand):
(WebKit::WebPage::completeSyntheticClick):
(WebKit::WebPage::handleDoubleTapForDoubleClickAtPoint):
(WebKit::WebPage::requestAdditionalItemsForDragSession):
(WebKit::WebPage::inspectorNodeSearchMovedToPosition):
(WebKit::WebPage::dispatchSyntheticMouseEventsForSelectionGesture):
* Source/WebKit/WebProcess/WebPage/mac/WebPageMac.mm:
(WebKit::WebPage::executeKeypressCommandsInternal):
(WebKit::WebPage::handleEditingKeyboardEvent):
* Source/WebKitLegacy/mac/DOM/DOMDocument.mm:
* Source/WebKitLegacy/mac/WebView/WebFrame.mm:
(-[WebFrame _dragSourceEndedAt:operation:]):
* Source/WebKitLegacy/mac/WebView/WebHTMLView.mm:
(currentKeyboardEvent):
(-[WebHTMLView _interpretKeyEvent:savingCommands:]):
* Source/WebKitLegacy/mac/WebView/WebPDFView.mm:
(-[WebPDFView PDFViewWillClickOnLink:withURL:]):

Canonical link: <a href="https://commits.webkit.org/259231@main">https://commits.webkit.org/259231@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/25f7b55200318846710ffebfa20c3c028f0d2cfc

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/104352 "1 style error") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/13435 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/37262 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/113568 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/173861 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/108276 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/14530 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/4350 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/96575 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/112608 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/110119 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/11196 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/94259 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/38828 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/93059 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/25870 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/80498 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/6795 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/27227 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/6925 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/3780 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/12948 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/46785 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/6365 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/8716 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

darinadler commented on Source/WebCore/bindings/js/ScriptController.h:
```diff
@@ -108,11 +108,11 @@ class ScriptController : public CanMakeWeakPtr<ScriptController> {
     // This asserts that URL argument is a JavaScript URL.
-    void executeJavaScriptURL(const URL&, RefPtr<SecurityOrigin>, ShouldReplaceDocumentIfJavaScriptURL, bool& didReplaceDocument);
-    void executeJavaScriptURL(const URL& url, RefPtr<SecurityOrigin> securityOrigin = nullptr, ShouldReplaceDocumentIfJavaScriptURL shouldReplaceDocumentIfJavaScriptURL = ReplaceDocumentIfJavaScriptURL)
+    void executeJavaScriptURL(const URL&, RefPtr<SecurityOrigin>&&, ShouldReplaceDocumentIfJavaScriptURL, bool& didReplaceDocument);
```
How is the added `&&` here related to the overall mission of the patch?

jernoble replied:
ScriptController.h doesn't include SecurityOrigin.h directly; it gets it by way of Document.h. So when I removed SecurityOrigin.h from Document.h, it caused a build error here.

When passing RefPtr<SecurityOrigin> by value, the copy constructor for RefPtr is called, which will call `ref()` on the SecurityOrigin, which necessitates having a definition for SecurityOrigin rather than a forward declaration. When passing by r-value reference, no methods on SecurityOrigin are called, so a forward declaration is fine.

I could have just #included SecurityOrigin.h here in ScriptController.h, or I could have moved the `executeJavaScriptURL()` method into the implementation file. But this felt like a better solution, since it allows the caller to eliminate a potentially unnecessary refcount churn, and it doesn't require including SecurityOrigin.h.

darinadler replied:
I would have moved the `executeJavaScriptURL` function into the implementation file. It’s really not all that helpful to have it inlined.

And since `executeJavaScriptURL` does not take ownership of the security origin but rather just uses it for a check, changing it to an rvalue reference seems like a step in the wrong direction. If it’s OK for our smart pointer security strategy, I would also consider change the argument to a raw pointer. If not, then the other alternatives above apply.

jernoble replied:
Okay, I'll undo this change and move it into the implementation file.

jernoble replied:
(Also, rather than a raw pointer, this could be two functions, one that takes a `const SecurityOrigin&` and one that doesn't, and it would be the client's responsibility to deal with possibly null values, and not the function's.)

darinadler commented on Source/WebCore/dom/Document.h:
```diff
@@ -2269,7 +2278,7 @@ class Document
     bool m_updateTitleTaskScheduled { false };
 
-    FocusTrigger m_latestFocusTrigger { FocusTrigger::Other };
+    FocusTrigger m_latestFocusTrigger;
```
Is there a way we can still initialize this here? Like maybe put `{ }` here?

jernoble replied:
I mean, you could, but would that be a good idea? The order of the enumeration values may change in the future, and it's not obvious that this call site would need to change as well. Also, zero may not be a valid enumeration value, and, so this could possibly lead to initializing m_latestFocusTrigger with an invalid value. I just checked and there is no error or warning thrown in this case.

darinadler replied:
I do think it’s a good idea.

Generally speaking, uninitialized memory is even worse than the things you mentioned. Can lead to unpredictable behavior and even security vulnerabilities. Reliably initializing to a value that we can then overwrite in the constructor if it’s not really a suitable default is mild by comparison.

Typical enumerations will have a zero-valued item, typically the first one, and in future we can try to make sure that first value is a good default for most situations, just as we do with a class when we decide what the default constructor does.

jernoble replied:
I see, you're interested in protecting against a programming error where someone fails to correctly initialize the instance variable in the constructor. I'd prefer that we'd protect against this with the compiler, but it looks like clang's `-Wuninitialized` doesn't catch this case (but GCC does), so maybe we can't rely on the compiler here.

Okay, so long as we re-initialize to a explicit value in the implementation file, it seems reasonable to zero-init in the header first.

mcatanzaro replied:
I agree with Darin: the `{ }` initialization in the class body is worth it. Uninitialized memory should be avoided at all costs, even if we expect it to be reinitialized later.

darinadler commented on Source/WebCore/dom/Document.h:
```diff
@@ -2343,7 +2352,7 @@ class Document
 
 #if ENABLE(DOM_AUDIO_SESSION)
-    DOMAudioSession::Type m_audioSessionType { DOMAudioSession::Type::Auto };
+    DOMAudioSessionType m_audioSessionType;
```
Same question.

darinadler commented on Source/WebCore/dom/DocumentInlines.h:
```diff
@@ -117,4 +121,7 @@ inline WebCoreOpaqueRoot Node::opaqueRoot() const
     return traverseToOpaqueRoot();
 }
 
+inline bool Document::wasLastFocusByClick() const { return m_latestFocusTrigger == FocusTrigger::Click; }
+
+
 } // namespace WebCore
```
Extra blank line here.

jernoble committed:
Refactor Document.h to reduce build times
https://bugs.webkit.org/show_bug.cgi?id=250887
rdar://104465840

Reviewed by Darin Adler.

Replace explicit header includes with forward declarations wherever possible.

For some headers,
this requires enumerations to be moved outside an owning class and into namespace scope so those
enumerations can be forward declared.

For CanvasBase.h, pull CanvasObserver.h into its own file with minimal header dependencies.

For ScriptController, eliminate the requirement for SecurityOrigin to be included by modifying
executeJavaScriptURL() to take an r-value reference to a RefPtr<SecurityOrigin>.

With these changes, the compile-time cost of including Document.h has been reduced by 75%.

* Source/WebCore/Modules/audiosession/DOMAudioSession.h:
* Source/WebCore/Modules/mediastream/CanvasCaptureMediaStreamTrack.h:
* Source/WebCore/Modules/screen-wake-lock/WakeLock.cpp:
* Source/WebCore/Modules/webxr/WebXRSession.cpp:
* Source/WebCore/Modules/webxr/WebXRWebGLLayer.h:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/bindings/js/ScriptController.cpp:
(WebCore::ScriptController::executeJavaScriptURL):
* Source/WebCore/bindings/js/ScriptController.h:
(WebCore::ScriptController::executeJavaScriptURL):
* Source/WebCore/dom/Document.cpp:
(WebCore::Document::setFocusedElement):
(WebCore::Document::hasListenerTypeForEventType const):
(WebCore::m_reportingScope): Deleted.
* Source/WebCore/dom/Document.h:
(WebCore::Document::setFocusedElement): Deleted.
(WebCore::Document::wasLastFocusByClick const): Deleted.
(WebCore::Document::clientOrigin const): Deleted.
* Source/WebCore/dom/DocumentInlines.h:
(WebCore::Document::clientOrigin const):
(WebCore::Document::wasLastFocusByClick const):
* Source/WebCore/dom/Element.cpp:
(WebCore::isForceEvent):
(WebCore::Element::dispatchMouseForceWillBegin):
* Source/WebCore/dom/FragmentScriptingPermission.h:
* Source/WebCore/dom/KeyboardEvent.cpp:
(WebCore::eventTypeForKeyboardEventType):
* Source/WebCore/dom/VisibilityState.h:
* Source/WebCore/html/CanvasBase.h:
(WebCore::CanvasObserver::isStyleCanvasImage const): Deleted.
* Source/WebCore/html/CanvasObserver.h: Copied from Source/WebCore/page/DisabledAdaptations.h.
(WebCore::CanvasObserver::isStyleCanvasImage const):
* Source/WebCore/page/Chrome.h:
* Source/WebCore/page/ContextMenuController.cpp:
(WebCore::ContextMenuController::showContextMenuAt):
* Source/WebCore/page/DebugPageOverlays.cpp:
(WebCore::InteractionRegionOverlay::mouseEvent):
* Source/WebCore/page/DisabledAdaptations.h:
* Source/WebCore/page/DragController.cpp:
(WebCore::createMouseEvent):
* Source/WebCore/page/EventHandler.cpp:
(WebCore::SyntheticTouchPoint::SyntheticTouchPoint):
(WebCore::SyntheticSingleTouchEvent::SyntheticSingleTouchEvent):
(WebCore::EventHandler::handleMouseForceEvent):
(WebCore::EventHandler::handlePasteGlobalSelection):
(WebCore::EventHandler::sendContextMenuEventForKey):
(WebCore::EventHandler::fakeMouseMoveEventTimerFired):
(WebCore::EventHandler::isKeyEventAllowedInFullScreen const):
(WebCore::EventHandler::internalKeyEvent):
(WebCore::EventHandler::handleDrag):
(WebCore::EventHandler::dispatchSyntheticTouchEventIfEnabled):
* Source/WebCore/page/ResourceUsageOverlay.cpp:
(WebCore::ResourceUsageOverlay::mouseEvent):
* Source/WebCore/page/ios/EventHandlerIOS.mm:
(WebCore::EventHandler::dispatchSimulatedTouchEvent):
(WebCore::EventHandler::tryToBeginDragAtPoint):
* Source/WebCore/page/mac/ImageOverlayControllerMac.mm:
(WebCore::ImageOverlayController::platformHandleMouseEvent):
* Source/WebCore/page/mac/ServicesOverlayController.mm:
(WebCore::ServicesOverlayController::mouseEvent):
* Source/WebCore/platform/KeyboardScrollingAnimator.cpp:
(WebCore::keyboardScrollingKeyForKeyboardEvent):
* Source/WebCore/platform/PlatformEvent.h:
(WebCore::PlatformEvent::type const):
(WebCore::PlatformEvent::PlatformEvent):
(): Deleted.
* Source/WebCore/platform/PlatformKeyboardEvent.h:
(WebCore::PlatformKeyboardEvent::PlatformKeyboardEvent):
* Source/WebCore/platform/PlatformMouseEvent.h:
(WebCore::PlatformMouseEvent::PlatformMouseEvent):
* Source/WebCore/platform/PlatformWheelEvent.cpp:
(WebCore::PlatformWheelEvent::createFromGesture):
* Source/WebCore/platform/PlatformWheelEvent.h:
(WebCore::PlatformWheelEvent::PlatformWheelEvent):
* Source/WebCore/platform/ios/KeyEventIOS.mm:
(WebCore::PlatformKeyboardEvent::disambiguateKeyDownEvent):
* Source/WebCore/platform/ios/PlatformEventFactoryIOS.mm:
(WebCore::mouseEventType):
(WebCore::PlatformWheelEventBuilder::PlatformWheelEventBuilder):
(WebCore::PlatformKeyboardEventBuilder::PlatformKeyboardEventBuilder):
(WebCore::touchEventType):
(WebCore::touchPhaseFromPlatformEventType):
* Source/WebCore/platform/ios/ScrollAnimatorIOS.mm:
(WebCore::ScrollAnimatorIOS::handleTouchEvent):
* Source/WebCore/platform/libwebrtc/LibWebRTCVPXVideoDecoder.cpp:
* Source/WebCore/platform/mac/KeyEventMac.mm:
(WebCore::PlatformKeyboardEvent::disambiguateKeyDownEvent):
* Source/WebCore/platform/mac/PlatformEventFactoryMac.mm:
(WebCore::mouseEventTypeForEvent):
(WebCore::PlatformMouseEventBuilder::PlatformMouseEventBuilder):
(WebCore::PlatformWheelEventBuilder::PlatformWheelEventBuilder):
(WebCore::PlatformKeyboardEventBuilder::PlatformKeyboardEventBuilder):
* Source/WebCore/rendering/style/StyleCanvasImage.h:
* Source/WebCore/rendering/style/StyleCrossfadeImage.h:
* Source/WebCore/xml/DOMParser.cpp:
* Source/WebKit/Shared/WebEventConversion.cpp:
(WebKit::platform):
(WebKit::WebKit2PlatformMouseEvent::WebKit2PlatformMouseEvent):
(WebKit::WebKit2PlatformWheelEvent::WebKit2PlatformWheelEvent):
(WebKit::WebKit2PlatformKeyboardEvent::WebKit2PlatformKeyboardEvent):
(WebKit::WebKit2PlatformTouchEvent::WebKit2PlatformTouchEvent):
(WebKit::WebKit2PlatformGestureEvent::WebKit2PlatformGestureEvent):
* Source/WebKit/UIProcess/RemoteLayerTree/mac/ScrollerPairMac.mm:
(WebKit::ScrollerPairMac::handleMouseEvent):
* Source/WebKit/WebProcess/Network/WebSocketChannel.cpp:
* Source/WebKit/WebProcess/WebPage/ios/WebPageIOS.mm:
(WebKit::dispatchSyntheticMouseMove):
(WebKit::WebPage::generateSyntheticEditingCommand):
(WebKit::WebPage::completeSyntheticClick):
(WebKit::WebPage::handleDoubleTapForDoubleClickAtPoint):
(WebKit::WebPage::requestAdditionalItemsForDragSession):
(WebKit::WebPage::inspectorNodeSearchMovedToPosition):
(WebKit::WebPage::dispatchSyntheticMouseEventsForSelectionGesture):
* Source/WebKit/WebProcess/WebPage/mac/WebPageMac.mm:
(WebKit::WebPage::executeKeypressCommandsInternal):
(WebKit::WebPage::handleEditingKeyboardEvent):
* Source/WebKitLegacy/mac/DOM/DOMDocument.mm:
* Source/WebKitLegacy/mac/WebView/WebFrame.mm:
(-[WebFrame _dragSourceEndedAt:operation:]):
* Source/WebKitLegacy/mac/WebView/WebHTMLView.mm:
(currentKeyboardEvent):
(-[WebHTMLView _interpretKeyEvent:savingCommands:]):
* Source/WebKitLegacy/mac/WebView/WebPDFView.mm:
(-[WebPDFView PDFViewWillClickOnLink:withURL:]):

Canonical link: https://commits.webkit.org/259231@main
